packages/now-build-utils/src/types.ts

```diff
@@ -320,3 +321,24 @@ export interface BuilderFunctions {
     excludeFiles?: string;
   };
 }
+
+export type Detector = (params: DetectorParameters) => Promise<DetectorResult>;
+
+export interface DetectorParameters {
+  fs: DetectorFilesystem;
+  detectors?: Detector[];
+  pkgDetectors?: Detector[];
+}
+
+export interface DetectorOutput {
+  buildCommand: string[];
+  buildDirectory: string;
+  buildEnv?: Env;
+  devCommand?: string[];
+  devEnv?: Env;
+  minNodeRange?: string;
+  routes?: Route[];
```

Comment:
Let's use the new routing properties here:

- `rewrites`
- `redirects`
- `headers`
- `trailingSlash`
- `cleanUrls`